core/state/snapshot/snapshot.go

```diff
@@ -689,7 +695,14 @@ func (t *Tree) Journal(root common.Hash) (common.Hash, error) {
 // Rebuild wipes all available snapshot data from the persistent database and
 // discard all caches and diff layers. Afterwards, it starts a new snapshot
 // generator with the given root hash.
-func (t *Tree) Rebuild(root common.Hash) {
+func (t *Tree) Rebuild(root common.Hash, async bool) {
+	var genPending chan struct{}
+	if !async {
```

Comment:
1. Why use defer and not just check async at the end of function?
(if using defer, I'd at least check that genPending is not nil)

2. why not use waitBuild for waiting?

Reply:
I used a defer because I wanted it to happen after the mutex was unlocked. I'll switch it to waitBuild though.

Reply:
I've removed the need for this change